ocp-templates/app-interface/backend.yaml

```diff
@@ -146,243 +145,41 @@ objects:
-# Scheduler
-# ===============================================
-- kind: DeploymentConfig
-  apiVersion: v1
-  metadata:
-    labels:
-      app: ${APP_NAME}
-    name: ibutsu-scheduler
-  spec:
-    replicas: 1
-    selector:
-      deploymentConfig: ibutsu-scheduler
-    strategy:
-      type: Rolling
-    template:
-      metadata:
-        labels:
-          app: ${APP_NAME}
-          deploymentConfig: ibutsu-scheduler
-      spec:
-        containers:
-        - env:
-          - name: POSTGRESQL_HOST
-            valueFrom:
-              secretKeyRef:
-                key: db.host
-                name: ibutsu-db
-          - name: POSTGRESQL_PORT
-            valueFrom:
-              secretKeyRef:
-                key: db.port
-                name: ibutsu-db
-          - name: POSTGRESQL_USER
-            valueFrom:
-              secretKeyRef:
-                key: db.user
-                name: ibutsu-db
-          - name: POSTGRESQL_PASSWORD
-            valueFrom:
-              secretKeyRef:
-                key: db.password
-                name: ibutsu-db
-          - name: POSTGRESQL_DATABASE
-            valueFrom:
-              secretKeyRef:
-                key: db.name
-                name: ibutsu-db
-          - name: REDIS_PASSWORD
-            valueFrom:
-              secretKeyRef:
-                key: db.password
-                name: redis
-                optional: true
-          - name: REDIS_HOSTNAME
-            valueFrom:
-              secretKeyRef:
-                key: db.endpoint
-                name: redis
-          - name: REDIS_PORT
-            valueFrom:
-              secretKeyRef:
-                key: db.port
-                name: redis
-          - name: FRONTEND_URL
-            value: https://${FRONTEND_ROUTE}
-          - name: BACKEND_URL
-            value: https://${BACKEND_ROUTE}
-          image: quay.io/cloudservices/ibutsu-scheduler:${IMAGE_TAG}
-          imagePullPolicy: IfNotPresent
-          name: ibutsu-scheduler
-          resources: {}
-          terminationMessagePath: /dev/termination-log
-          terminationMessagePolicy: File
-        dnsPolicy: ClusterFirst
-        restartPolicy: Always
-    triggers:
-    - type: ConfigChange
-# -----------------------------------------------
-- kind: Service
-  apiVersion: v1
-  metadata:
-    labels:
-      app: ${APP_NAME}
-    name: ibutsu-scheduler
+    name: ibutsu-backend
+    annotations:
+      description: "A route to the backend"
   spec:
-    ports:
-    - port: 8080
-      targetPort: 8080
-    selector:
-      deploymentConfig: ibutsu-scheduler
-
-# ===============================================
-# Worker
-# ===============================================
-- kind: DeploymentConfig
+    host: ${BACKEND_ROUTE}
+    to:
+      kind: Service
+      name: ibutsu-backend
+    tls:
+      insecureEdgeTerminationPolicy: Redirect
+      termination: edge
+# ------------------------------------------------
+- kind: Secret
```

Comment:
Just FYI: Most apps would have secret like this inserted inserted into the namespace config via vault but with a special syntax, app-interface does support passing vault creds into the template as parameter values -- so we can make this work.

Reply:
I don't mind which way we do it, Vault sounds like a good option to me.